src/dbg/thread.cpp

```diff
@@ -37,9 +37,19 @@ void ThreadCreate(CREATE_THREAD_DEBUG_INFO* CreateThread)
     // Duplicate the debug thread handle -> thread handle
     DuplicateHandle(GetCurrentProcess(), CreateThread->hThread, GetCurrentProcess(), &curInfo.Handle, 0, FALSE, DUPLICATE_SAME_ACCESS);
 
+    typedef HRESULT(WINAPI * GETTHREADDESCRIPTION)(HANDLE hThread, PWSTR * ppszThreadDescription);
+    static GETTHREADDESCRIPTION _GetThreadDescription = (GETTHREADDESCRIPTION)GetProcAddress(GetModuleHandleW(L"kernel32.dll"), "GetThreadDescription");
+    PWSTR threadDescription;
+
     // The first thread (#0) is always the main program thread
     if(curInfo.ThreadNumber <= 0)
         strcpy_s(curInfo.threadName, GuiTranslateText(QT_TRANSLATE_NOOP("DBG", "Main Thread")));
+    else if(_GetThreadDescription && SUCCEEDED(_GetThreadDescription(curInfo.Handle, &threadDescription)))
+    {
+        if(threadDescription[0])
+            strcpy_s(curInfo.threadName, StringUtils::Utf16ToUtf8(threadDescription).c_str());

```

Comment:
This could overflow and crash x64dbg. Should probably be `strcpy_s(curInfo.threadName, StringUtils::Utf16ToUtf8(threadDescription).c_str(), _TRUNCATE);`